receieved "error reading from server" while trying to refresh

**Describe the bug**
While trying to remove a (git-lfs) repository from Artifactory via the terraform provider, I started to receive the message:
```
│ Error: Unable to Refresh Resource
│
│ An unexpected error occurred while attempting to refresh resource state. Please retry the operation or report this issue to the provider developers.
│
│ Error: Bad Request
```

If I enable `TF_LOG` I see the following:
```
2025-02-06T15:57:57.080-0500 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2025-02-06T15:57:57.080-0500 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2025-02-06T15:57:57.081-0500 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/jfrog/platform/1.19.1/darwin_arm64/terraform-provider-platform_v1.19.1 pid=56367
2025-02-06T15:57:57.081-0500 [DEBUG] provider: plugin exited
2025-02-06T15:57:57.081-0500 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/jfrog/artifactory/12.8.0/darwin_arm64/terraform-provider-artifactory_v12.8.0 pid=56368
2025-02-06T15:57:57.081-0500 [DEBUG] provider: plugin exited
```

Artifactory version:
```
jf rt curl api/system/version
{
  "version" : "7.98.14",
  "revision" : "79814900",
  "servicesVersions" : {
    "package_handler_version" : ""
  },
  "addons" : [ "build", "docker", "helmoci", "oci", "packages-archive", "vagrant", "replication", "filestore", "curation", "plugins", "worker", "gems", "composer", "npm", "bower", "nuget", "debian", "opkg", "rpm", "cocoapods", "conan", "vcs", "release-bundle", "jf-connect", "jf-event", "replicator", "keys", "alpine", "analytics", "cargo", "chef", "federated", "git", "observability", "onboarding", "pub", "rest", "swift", "lead-artifact-detector", "conda", "terraform", "tracker", "license", "puppet", "ldap", "sso", "layouts", "properties", "search", "securityresourceaddon", "filtered-resources", "p2", "watch", "webstart", "support", "xray", "retention", "policies", "archive" ],
  "license" : "03b04c9dc2fa519251bec2725a7506099c7a86732",
  "entitlements" : {
    "EVENT_BASED_PULL_REPLICATION" : false,
    "SMART_REMOTE_TARGET_FOR_EDGE" : false,
    "REPO_REPLICATION" : true,
    "MULTIPUSH_REPLICATION" : false
  }
}
```

Terraform Version:
```
tf version
Terraform v1.5.7
on darwin_arm64
+ provider registry.terraform.io/hashicorp/null v3.2.3
+ provider registry.terraform.io/hashicorp/random v3.6.3
+ provider registry.terraform.io/jfrog/artifactory v12.8.0
+ provider registry.terraform.io/jfrog/platform v1.19.1

Your version of Terraform is out of date! The latest version
is 1.10.5. You can update by downloading from https://www.terraform.io/downloads.html
```
**Requirements for and issue**
- [X] A description of the bug
- [-] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue). **If this is not supplied, this issue will likely be closed without any effort expended.**
- [X] Your version of artifactory (you can `curl` it at `$host/artifactory/api/system/version`
- [X] Your version of terraform
- [X] Your version of terraform provider

**Expected behavior**
This is from a `terraform refresh`, I would expect this to update the local state file.

**Additional context**
N/A

@alexhung , I'll was trying to get one for you but I think this was actually a result of a corrupted terraform state file.

I say that because while debugging further, I found this log entry:
```
2025-02-07T07:46:23.589-0500 [ERROR] provider.terraform-provider-artifactory_v12.8.0: Response contains error diagnostic: tf_req_id=9e1f8df4-5ffa-4191-960b-e4e8ff7b5ab1 @caller=github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov6/internal/diag/diagnostics.go:58 diagnostic_detail="An unexpected error occurred while attempting to refresh resource state. Please retry the operation or report this issue to the provider developers.

Error: Bad Request" tf_proto_version=6.7 tf_provider_addr=jfrog/artifactory @module=sdk.proto diagnostic_severity=ERROR diagnostic_summary="Unable to Refresh Resource" tf_resource_type=artifactory_local_gitlfs_repository tf_rpc=ReadResource timestamp=2025-02-07T07:46:23.589-0500
2025-02-07T07:46:23.589-0500 [ERROR] vertex "artifactory_local_gitlfs_repository.git-lfs-local" error: Unable to Refresh Resource
2025-02-07T07:46:23.589-0500 [ERROR] vertex "artifactory_local_gitlfs_repository.git-lfs-local (expand)" error: Unable to Refresh Resource
```

The repo (`git-lfs-local`) did exist, but was deleted in a previous run of terraform, however that change to delete the repository didn't seem to get committed to the terraform state file so runs of `terraform plan` were trying to refresh a state of a repository that had been deleted.

I manually created the repo in artifactory, then re-ran `terraform plan` and it showed that this repo was marked for deletion. This seems like an odd bug.

I was able to create a some steps to reproduce (this is not how I got the error originally but this is how I can reproduce the error message):

1) run terraform code at https://gist.github.com/madelaney/8fca799a78ba37de08ec3c4f42913d54
2) in artifactory, delete the newly created repository
3) run any `terraform plan` or `terraform apply`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,4 +1,4 @@
-## 12.8.4 (Februry 7, 2025)
+## 12.8.4 (Februry 7, 2025). Tested on Artifactory 7.104.6 with Terraform 1.10.5 and OpenTofu 1.9.0
 
 BUG FIXES:
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,9 @@
+## 12.8.4 (Februry 7, 2025)
+
+BUG FIXES:
+
+* resource/artifactory_\*\_repository: Fix error when provider tries to refresh the resource for repository that is no longer existed on Artifactory. Issue: [#1189](https://github.com/jfrog/terraform-provider-artifactory/issues/1189) PR: [#1190](https://github.com/jfrog/terraform-provider-artifactory/pull/1190)
+
 ## 12.8.3 (January 28, 2025). Tested on Artifactory 7.98.14 with Terraform 1.10.5 and OpenTofu 1.9.0
 
 IMPROVEMENTS:

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -49,7 +49,7 @@ install: clean build
 		# mv -v dist/terraform-provider-${PRODUCT}_${LINUX_GORELEASER_ARCH}/terraform-provider-${PRODUCT}_v${NEXT_PROVIDER_VERSION}* ${LINUX_BUILD_PATH} && \
 
 clean:
-	rm -fR dist terraform.d/ .terraform terraform.tfstate* terraform.d/ .terraform.lock.hcl
+	rm -fR dist terraform.d/ .terraform terraform.tfstate* .terraform.lock.hcl
 
 release:
 	@git tag ${NEXT_PROVIDER_VERSION} && git push --mirror

--- a/pkg/artifactory/datasource/repository/remote/datasource_artifactory_remote_bower_repository.go
+++ b/pkg/artifactory/datasource/repository/remote/datasource_artifactory_remote_bower_repository.go
@@ -18,7 +18,7 @@ type BowerRemoteRepo struct {
 
 var bowerSchema = lo.Assign(
 	remote.BaseSchema,
-	remote.VcsRemoteRepoSchemaSDKv2,
+	VcsRemoteRepoSchemaSDKv2,
 	map[string]*schema.Schema{
 		"bower_registry_url": {
 			Type:         schema.TypeString,

--- a/pkg/artifactory/datasource/repository/remote/datasource_artifactory_remote_cocoapods_repository.go
+++ b/pkg/artifactory/datasource/repository/remote/datasource_artifactory_remote_cocoapods_repository.go
@@ -18,7 +18,7 @@ type CocoapodsRemoteRepo struct {
 
 var cocoapodsSchema = lo.Assign(
 	remote.BaseSchema,
-	remote.VcsRemoteRepoSchemaSDKv2,
+	VcsRemoteRepoSchemaSDKv2,
 	map[string]*schema.Schema{
 		"pods_specs_repo_url": {
 			Type:         schema.TypeString,

--- a/pkg/artifactory/datasource/repository/remote/datasource_artifactory_remote_composer_repository.go
+++ b/pkg/artifactory/datasource/repository/remote/datasource_artifactory_remote_composer_repository.go
@@ -18,7 +18,7 @@ type ComposerRemoteRepo struct {
 
 var composerSchema = lo.Assign(
 	remote.BaseSchema,
-	remote.VcsRemoteRepoSchemaSDKv2,
+	VcsRemoteRepoSchemaSDKv2,
 	map[string]*schema.Schema{
 		"composer_registry_url": {
 			Type:         schema.TypeString,

--- a/pkg/artifactory/datasource/repository/remote/datasource_artifactory_remote_vcs_repository.go
+++ b/pkg/artifactory/datasource/repository/remote/datasource_artifactory_remote_vcs_repository.go
@@ -17,7 +17,7 @@ type VcsRemoteRepo struct {
 
 var VCSSchema = lo.Assign(
 	remote.BaseSchema,
-	remote.VcsRemoteRepoSchemaSDKv2,
+	VcsRemoteRepoSchemaSDKv2,
 	map[string]*schema.Schema{
 		"max_unique_snapshots": {
 			Type:     schema.TypeInt,

--- a/pkg/artifactory/datasource/repository/remote/remote.go
+++ b/pkg/artifactory/datasource/repository/remote/remote.go
@@ -2,6 +2,7 @@ package remote
 
 import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 	"github.com/jfrog/terraform-provider-artifactory/v12/pkg/artifactory/resource/repository/remote"
 )
 
@@ -12,4 +13,20 @@ var getSchema = func(schemas map[int16]map[string]*schema.Schema) map[string]*sc
 	s["url"].Optional = true
 
 	return s
+}
+
+var VcsRemoteRepoSchemaSDKv2 = map[string]*schema.Schema{
+	"vcs_git_provider": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		Default:          "GITHUB",
+		ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"GITHUB", "BITBUCKET", "OLDSTASH", "STASH", "ARTIFACTORY", "CUSTOM"}, false)),
+		Description:      `Artifactory supports proxying the following Git providers out-of-the-box: GitHub or a remote Artifactory instance. Default value is "GITHUB".`,
+	},
+	"vcs_git_download_url": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotEmpty),
+		Description:      `This attribute is used when vcs_git_provider is set to 'CUSTOM'. Provided URL will be used as proxy.`,
+	},
 }

--- a/pkg/artifactory/resource/configuration/resource_artifactory_archive_policy_test.go
+++ b/pkg/artifactory/resource/configuration/resource_artifactory_archive_policy_test.go
@@ -3,7 +3,9 @@ package configuration_test
 import (
 	"fmt"
 	"net/http"
+	"os"
 	"regexp"
+	"strings"
 	"testing"
 
 	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
@@ -139,6 +141,11 @@ func TestAccArchivePolicy_invalid_conditions(t *testing.T) {
 }
 
 func TestAccArchivePolicy_full(t *testing.T) {
+	archivePolicyEnabled := os.Getenv("JFROG_ARCHIVE_POLICY_ENABLED")
+	if strings.ToLower(archivePolicyEnabled) != "true" {
+		t.Skipf("JFROG_ARCHIVE_POLICY_ENABLED env var is not set to 'true'")
+	}
+
 	client := acctest.GetTestResty(t)
 	version, err := util.GetArtifactoryVersion(client)
 	if err != nil {
@@ -325,6 +332,11 @@ func TestAccArchivePolicy_full(t *testing.T) {
 }
 
 func TestAccArchivePolicy_with_project_key(t *testing.T) {
+	archivePolicyEnabled := os.Getenv("JFROG_ARCHIVE_POLICY_ENABLED")
+	if strings.ToLower(archivePolicyEnabled) != "true" {
+		t.Skipf("JFROG_ARCHIVE_POLICY_ENABLED env var is not set to 'true'")
+	}
+
 	client := acctest.GetTestResty(t)
 	version, err := util.GetArtifactoryVersion(client)
 	if err != nil {

--- a/pkg/artifactory/resource/configuration/resource_artifactory_package_cleanup_policy.go
+++ b/pkg/artifactory/resource/configuration/resource_artifactory_package_cleanup_policy.go
@@ -232,7 +232,7 @@ type PackageCleanupPolicySearchCriteriaAPIModel struct {
 	ExcludedRepos                *[]string `json:"excludedRepos,omitempty"`
 	IncludedPackages             []string  `json:"includedPackages"`
 	ExcludedPackages             *[]string `json:"excludedPackages,omitempty"`
-	IncludeAllProjects           *bool     `json:"includeAllProjects,omitempty"`
+	IncludeAllProjects           *bool     `json:"includeAllProjects"`
 	IncludedProjects             *[]string `json:"includedProjects,omitempty"`
 	CreatedBeforeInMonths        *int64    `json:"createdBeforeInMonths,omitempty"`
 	LastDownloadedBeforeInMonths *int64    `json:"lastDownloadedBeforeInMonths,omitempty"`
@@ -454,6 +454,8 @@ var cleanupPolicySchemaV1 = lo.Assign(
 				},
 				"include_all_projects": schema.BoolAttribute{
 					Optional:    true,
+					Computed:    true,
+					Default:     booldefault.StaticBool(false),
 					Description: "Set this to `true` if you want the policy to run on all projects on the platform.",
 				},
 				"included_projects": schema.SetAttribute{

--- a/pkg/artifactory/resource/configuration/resource_artifactory_package_cleanup_policy_test.go
+++ b/pkg/artifactory/resource/configuration/resource_artifactory_package_cleanup_policy_test.go
@@ -441,6 +441,12 @@ func TestAccPackageCleanupPolicy_with_project_key(t *testing.T) {
 		key             = "{{ .repoName }}"
 		tag_retention   = 3
 		max_unique_tags = 5
+
+		lifecycle {
+			ignore_changes = [
+				project_key,
+			]
+		}
 	}
 
 	resource "project" "myproject" {
@@ -455,6 +461,11 @@ func TestAccPackageCleanupPolicy_with_project_key(t *testing.T) {
 		max_storage_in_gibibytes   = 10
 		block_deployments_on_limit = false
 		email_notification         = true
+	}
+
+	resource "project_repository" "{{ .repoName }}" {
+		project_key = project.myproject.key
+		key         = artifactory_local_docker_v2_repository.{{ .repoName }}.key
 	}
 
 	resource "artifactory_package_cleanup_policy" "{{ .policyName }}" {
@@ -475,6 +486,8 @@ func TestAccPackageCleanupPolicy_with_project_key(t *testing.T) {
 			created_before_in_months = 1
 			last_downloaded_before_in_months = 6
 		}
+
+		depends_on = [project_repository.{{ .repoName }}]
 	}`
 
 	updatedTemp := `

--- a/pkg/artifactory/resource/repository/local/resource_artifactory_local_machinelearning_repository.go
+++ b/pkg/artifactory/resource/repository/local/resource_artifactory_local_machinelearning_repository.go
@@ -4,9 +4,12 @@ import (
 	"context"
 	"reflect"
 
+	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/resource"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 	"github.com/jfrog/terraform-provider-artifactory/v12/pkg/artifactory/resource/repository"
+	"github.com/samber/lo"
 )
 
 func NewMachineLearningLocalRepositoryResource() resource.Resource {
@@ -14,8 +17,8 @@ func NewMachineLearningLocalRepositoryResource() resource.Resource {
 		localResource: NewLocalRepositoryResource(
 			repository.MachineLearningType,
 			"Machine Learning",
-			reflect.TypeFor[LocalResourceModel](),
-			reflect.TypeFor[LocalAPIModel](),
+			reflect.TypeFor[LocalMachineLearningResourceModel](),
+			reflect.TypeFor[LocalMachineLearningAPIModel](),
 		),
 	}
 }
@@ -24,9 +27,84 @@ type MachineLearningLocalRepositoryResource struct {
 	localResource
 }
 
+type LocalMachineLearningResourceModel struct {
+	LocalResourceModel
+}
+
+func (r *LocalMachineLearningResourceModel) GetCreateResourcePlanData(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, r)...)
+}
+
+func (r LocalMachineLearningResourceModel) SetCreateResourceStateData(ctx context.Context, resp *resource.CreateResponse) {
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &r)...)
+}
+
+func (r *LocalMachineLearningResourceModel) GetReadResourceStateData(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	// Read Terraform state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, r)...)
+}
+
+func (r LocalMachineLearningResourceModel) SetReadResourceStateData(ctx context.Context, resp *resource.ReadResponse) {
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &r)...)
+}
+
+func (r *LocalMachineLearningResourceModel) GetUpdateResourcePlanData(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Read Terraform state data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, r)...)
+}
+
+func (r *LocalMachineLearningResourceModel) GetUpdateResourceStateData(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Read Terraform state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, r)...)
+}
+
+func (r LocalMachineLearningResourceModel) SetUpdateResourceStateData(ctx context.Context, resp *resource.UpdateResponse) {
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &r)...)
+}
+
+func (r LocalMachineLearningResourceModel) ToAPIModel(ctx context.Context, packageType string) (interface{}, diag.Diagnostics) {
+	diags := diag.Diagnostics{}
+
+	model, d := r.LocalResourceModel.ToAPIModel(ctx, packageType)
+	if d != nil {
+		diags.Append(d...)
+	}
+
+	localAPIModel := model.(LocalAPIModel)
+	localAPIModel.RepoLayoutRef = r.RepoLayoutRef.ValueString()
+
+	return LocalMachineLearningAPIModel{
+		LocalAPIModel: localAPIModel,
+	}, diags
+}
+
+func (r *LocalMachineLearningResourceModel) FromAPIModel(ctx context.Context, apiModel interface{}) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+
+	model := apiModel.(*LocalMachineLearningAPIModel)
+
+	r.LocalResourceModel.FromAPIModel(ctx, model.LocalAPIModel)
+
+	r.RepoLayoutRef = types.StringValue(model.RepoLayoutRef)
+
+	return diags
+}
+
+type LocalMachineLearningAPIModel struct {
+	LocalAPIModel
+}
+
 func (r *MachineLearningLocalRepositoryResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	attributes := lo.Assign(
+		LocalAttributes,
+		repository.RepoLayoutRefAttribute(r.Rclass, r.PackageType),
+	)
 	resp.Schema = schema.Schema{
-		Attributes:  LocalAttributes,
+		Attributes:  attributes,
 		Description: r.Description,
 	}
 }

--- a/pkg/artifactory/resource/repository/local/resource_artifactory_local_machinelearning_repository_test.go
+++ b/pkg/artifactory/resource/repository/local/resource_artifactory_local_machinelearning_repository_test.go
@@ -6,6 +6,7 @@ import (
 
 	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
 	"github.com/jfrog/terraform-provider-artifactory/v12/pkg/acctest"
+	"github.com/jfrog/terraform-provider-artifactory/v12/pkg/artifactory/resource/repository"
 	"github.com/jfrog/terraform-provider-shared/testutil"
 	"github.com/jfrog/terraform-provider-shared/util"
 )
@@ -53,6 +54,10 @@ func TestAccLocalMachineLearningRepository(t *testing.T) {
 					resource.TestCheckResourceAttr(fqrn, "key", name),
 					resource.TestCheckResourceAttr(fqrn, "blacked_out", strconv.FormatBool(params["blacked_out"].(bool))),
 					resource.TestCheckResourceAttr(fqrn, "xray_index", strconv.FormatBool(params["xray_index"].(bool))),
+					resource.TestCheckResourceAttr(fqrn, "repo_layout_ref", func() string {
+						r, _ := repository.GetDefaultRepoLayoutRef("local", repository.MachineLearningType)
+						return r
+					}()), //Check to ensure repository layout is set as per default even when it is not passed.
 					resource.TestCheckResourceAttr(fqrn, "property_sets.#", "1"),
 					resource.TestCheckResourceAttr(fqrn, "property_sets.0", params["property_set"].(string)),
 					resource.TestCheckResourceAttr(fqrn, "archive_browsing_enabled", strconv.FormatBool(params["archive_browsing_enabled"].(bool))),

--- a/pkg/artifactory/resource/repository/remote/remote.go
+++ b/pkg/artifactory/resource/repository/remote/remote.go
@@ -314,6 +314,25 @@ type RemoteAPIModel struct {
 	DisableURLNormalization           bool                    `json:"disableUrlNormalization"`
 }
 
+type ContentSynchronisation struct {
+	Enabled    bool                             `json:"enabled"`
+	Statistics ContentSynchronisationStatistics `json:"statistics"`
+	Properties ContentSynchronisationProperties `json:"properties"`
+	Source     ContentSynchronisationSource     `json:"source"`
+}
+
+type ContentSynchronisationStatistics struct {
+	Enabled bool `hcl:"statistics_enabled" json:"enabled"`
+}
+
+type ContentSynchronisationProperties struct {
+	Enabled bool `hcl:"properties_enabled" json:"enabled"`
+}
+
+type ContentSynchronisationSource struct {
+	OriginAbsenceDetection bool `hcl:"source_origin_absence_detection" json:"originAbsenceDetection"`
+}
+
 type vcsAPIModel struct {
 	GitProvider    *string `json:"vcsGitProvider,omitempty"`
 	GitDownloadURL *string `json:"vcsGitDownloadUrl,omitempty"`
@@ -710,24 +729,6 @@ func (r remoteResource) ValidateConfig(ctx context.Context, req resource.Validat
 }
 
 // SDKv2
-type ContentSynchronisation struct {
-	Enabled    bool                             `json:"enabled"`
-	Statistics ContentSynchronisationStatistics `json:"statistics"`
-	Properties ContentSynchronisationProperties `json:"properties"`
-	Source     ContentSynchronisationSource     `json:"source"`
-}
-
-type ContentSynchronisationStatistics struct {
-	Enabled bool `hcl:"statistics_enabled" json:"enabled"`
-}
-
-type ContentSynchronisationProperties struct {
-	Enabled bool `hcl:"properties_enabled" json:"enabled"`
-}
-
-type ContentSynchronisationSource struct {
-	OriginAbsenceDetection bool `hcl:"source_origin_absence_detection" json:"originAbsenceDetection"`
-}
 
 type RepositoryRemoteBaseParams struct {
 	Key                               string                  `json:"key,omitempty"`
@@ -1129,22 +1130,6 @@ var GetSchemas = func(s map[string]*sdkv2_schema.Schema) map[int16]map[string]*s
 	}
 }
 
-var VcsRemoteRepoSchemaSDKv2 = map[string]*sdkv2_schema.Schema{
-	"vcs_git_provider": {
-		Type:             sdkv2_schema.TypeString,
-		Optional:         true,
-		Default:          "GITHUB",
-		ValidateDiagFunc: sdkv2_validator.ToDiagFunc(sdkv2_validator.StringInSlice([]string{"GITHUB", "BITBUCKET", "OLDSTASH", "STASH", "ARTIFACTORY", "CUSTOM"}, false)),
-		Description:      `Artifactory supports proxying the following Git providers out-of-the-box: GitHub or a remote Artifactory instance. Default value is "GITHUB".`,
-	},
-	"vcs_git_download_url": {
-		Type:             sdkv2_schema.TypeString,
-		Optional:         true,
-		ValidateDiagFunc: sdkv2_validator.ToDiagFunc(sdkv2_validator.StringIsNotEmpty),
-		Description:      `This attribute is used when vcs_git_provider is set to 'CUSTOM'. Provided URL will be used as proxy.`,
-	},
-}
-
 func JavaSchema(packageType string, suppressPom bool) map[string]*sdkv2_schema.Schema {
 	return lo.Assign(
 		BaseSchema,
@@ -1209,99 +1194,4 @@ func JavaSchema(packageType string, suppressPom bool) map[string]*sdkv2_schema.S
 		},
 		repository.RepoLayoutRefSDKv2Schema(Rclass, packageType),
 	)
-}
-
-func UnpackBaseRemoteRepo(s *sdkv2_schema.ResourceData, packageType string) RepositoryRemoteBaseParams {
-	d := &utilsdk.ResourceData{ResourceData: s}
-
-	repo := RepositoryRemoteBaseParams{
-		Rclass:                            Rclass,
-		Key:                               d.GetString("key", false),
-		ProjectKey:                        d.GetString("project_key", false),
-		ProjectEnvironments:               d.GetSet("project_environments"),
-		PackageType:                       packageType, // must be set independently
-		Url:                               d.GetString("url", false),
-		Username:                          d.GetString("username", false),
-		Password:                          d.GetString("password", false),
-		Proxy:                             d.GetString("proxy", false),
-		DisableProxy:                      d.GetBool("disable_proxy", false),
-		Description:                       d.GetString("description", false),
-		Notes:                             d.GetString("notes", false),
-		IncludesPattern:                   d.GetString("includes_pattern", false),
-		ExcludesPattern:                   d.GetString("excludes_pattern", false),
-		RepoLayoutRef:                     d.GetString("repo_layout_ref", false),
-		RemoteRepoLayoutRef:               d.GetString("remote_repo_layout_ref", false),
-		HardFail:                          d.GetBoolRef("hard_fail", false),
-		Offline:                           d.GetBoolRef("offline", false),
-		BlackedOut:                        d.GetBoolRef("blacked_out", false),
-		XrayIndex:                         d.GetBool("xray_index", false),
-		DownloadRedirect:                  d.GetBool("download_direct", false),
-		CdnRedirect:                       d.GetBool("cdn_redirect", false),
-		QueryParams:                       d.GetString("query_params", false),
-		StoreArtifactsLocally:             d.GetBoolRef("store_artifacts_locally", false),
-		SocketTimeoutMillis:               d.GetInt("socket_timeout_millis", false),
-		LocalAddress:                      d.GetString("local_address", false),
-		RetrievalCachePeriodSecs:          d.GetInt("retrieval_cache_period_seconds", false),
-		MissedRetrievalCachePeriodSecs:    d.GetInt("missed_cache_period_seconds", false),
-		MetadataRetrievalTimeoutSecs:      d.GetInt("metadata_retrieval_timeout_secs", false),
-		UnusedArtifactsCleanupPeriodHours: d.GetInt("unused_artifacts_cleanup_period_hours", false),
-		AssumedOfflinePeriodSecs:          d.GetInt("assumed_offline_period_secs", false),
-		ShareConfiguration:                d.GetBoolRef("share_configuration", false),
-		SynchronizeProperties:             d.GetBoolRef("synchronize_properties", false),
-		BlockMismatchingMimeTypes:         d.GetBoolRef("block_mismatching_mime_types", false),
-		PropertySets:                      d.GetSet("property_sets"),
-		AllowAnyHostAuth:                  d.GetBoolRef("allow_any_host_auth", false),
-		EnableCookieManagement:            d.GetBoolRef("enable_cookie_management", false),
-		BypassHeadRequests:                d.GetBoolRef("bypass_head_requests", false),
-		ClientTLSCertificate:              d.GetString("client_tls_certificate", false),
-		PriorityResolution:                d.GetBool("priority_resolution", false),
-		ListRemoteFolderItems:             d.GetBool("list_remote_folder_items", false),
-		MismatchingMimeTypeOverrideList:   d.GetString("mismatching_mime_types_override_list", false),
-		DisableURLNormalization:           d.GetBool("disable_url_normalization", false),
-		ArchiveBrowsingEnabled:            d.GetBoolRef("archive_browsing_enabled", false),
-	}
-
-	if v, ok := d.GetOk("content_synchronisation"); ok {
-		contentSynchronisationConfig := v.([]interface{})[0].(map[string]interface{})
-		enabled := contentSynchronisationConfig["enabled"].(bool)
-		statisticsEnabled := contentSynchronisationConfig["statistics_enabled"].(bool)
-		propertiesEnabled := contentSynchronisationConfig["properties_enabled"].(bool)
-		sourceOriginAbsenceDetection := contentSynchronisationConfig["source_origin_absence_detection"].(bool)
-		repo.ContentSynchronisation = &ContentSynchronisation{
-			Enabled: enabled,
-			Statistics: ContentSynchronisationStatistics{
-				Enabled: statisticsEnabled,
-			},
-			Properties: ContentSynchronisationProperties{
-				Enabled: propertiesEnabled,
-			},
-			Source: ContentSynchronisationSource{
-				OriginAbsenceDetection: sourceOriginAbsenceDetection,
-			},
-		}
-	}
-	return repo
-}
-
-func UnpackVcsRemoteRepo(s *sdkv2_schema.ResourceData) RepositoryVcsParams {
-	d := &utilsdk.ResourceData{ResourceData: s}
-	return RepositoryVcsParams{
-		VcsGitProvider:    d.GetString("vcs_git_provider", false),
-		VcsGitDownloadUrl: d.GetString("vcs_git_download_url", false),
-	}
-}
-
-func UnpackJavaRemoteRepo(s *sdkv2_schema.ResourceData, repoType string) JavaRemoteRepo {
-	d := &utilsdk.ResourceData{ResourceData: s}
-	return JavaRemoteRepo{
-		RepositoryRemoteBaseParams:   UnpackBaseRemoteRepo(s, repoType),
-		FetchJarsEagerly:             d.GetBool("fetch_jars_eagerly", false),
-		FetchSourcesEagerly:          d.GetBool("fetch_sources_eagerly", false),
-		RemoteRepoChecksumPolicyType: d.GetString("remote_repo_checksum_policy_type", false),
-		HandleReleases:               d.GetBool("handle_releases", false),
-		HandleSnapshots:              d.GetBool("handle_snapshots", false),
-		SuppressPomConsistencyChecks: d.GetBool("suppress_pom_consistency_checks", false),
-		RejectInvalidJars:            d.GetBool("reject_invalid_jars", false),
-		MaxUniqueSnapshots:           d.GetInt("max_unique_snapshots", false),
-	}
 }

--- a/pkg/artifactory/resource/repository/repository.go
+++ b/pkg/artifactory/resource/repository/repository.go
@@ -260,9 +260,9 @@ func (r *BaseResource) Read(ctx context.Context, req resource.ReadRequest, resp 
 		return
 	}
 
-	// Treat HTTP 404 Not Found status as a signal to recreate resource
+	// Treat HTTP 400 Bad Request or 404 Not Found status as a signal to recreate resource
 	// and return early
-	if response.StatusCode() == http.StatusNotFound {
+	if response.StatusCode() == http.StatusBadRequest || response.StatusCode() == http.StatusNotFound {
 		resp.State.RemoveResource(ctx)
 		return
 	}

--- a/pkg/artifactory/resource/repository/repository.go
+++ b/pkg/artifactory/resource/repository/repository.go
@@ -616,6 +616,7 @@ var CompressionFormatsAttribute = map[string]schema.Attribute{
 	},
 }
 
+// SDKv2
 var BaseSchemaV1 = map[string]*sdkv2_schema.Schema{
 	"key": {
 		Type:             sdkv2_schema.TypeString,

